File: glance-c/src/clipboard.c
/* clipboard.c — system clipboard via pbcopy (macOS). */
#include "clipboard.h"

#include <stdio.h>

int clip_copy(const char *text, size_t len) {
    FILE *p = popen("pbcopy", "w");
    if (!p) return -1;
    fwrite(text, 1, len, p);
    return pclose(p) == 0 ? 0 : -1;
}

File: legacy-go/internal/app/clipboard.go
package app

import "github.com/atotto/clipboard"

// copyToClipboard writes text to the system clipboard. On macOS this shells
// out to pbcopy via atotto/clipboard, so the copied text is available to any
// other application — i.e. it survives outside glance.
func copyToClipboard(text string) error {
	return clipboard.WriteAll(text)
}
